File: assign3/ruby/test/pizza_test.rb
require_relative "../src/pizza"
require "test/unit"
 
class TestPizza < Test::Unit::TestCase

	def test_canary
		assert(true)
	end

	def test_create_pizza_with_name
		pizz = create pizza
		assert_equal("Pizza with no layer", pizz.to_s)
	end
	
	def test_create_pizza_with_many_layers
		pizz = create pizza do
			spread sauce
			spread cheese
			toppings meat, onions
			spread cheese
			bake
		end
		assert_equal("Pizza with 5 layers: sauce, cheese, meat, onions, cheese", pizz.to_s)
	end
end

File: assign3/scala/test/pizza/PizzaTest.scala
package pizza.PizzaDSL

import org.scalatest.FunSuite
import org.scalatest.junit.JUnitRunner
import org.junit.runner.RunWith

@RunWith(classOf[JUnitRunner])
class PizzaTest extends FunSuite {

	test("Create pizza") {
		val pizz = (create a pizza)
		assert("Pizza with no layer" == pizz.toString)
	}
	
	test("Create pizza with layers") {
		val pizz = (create a pizza 
					spread sauce 
					toppings onions)
		assert("Pizza with 2 layers: sauce, onions" == pizz.toString)
	}
	
	test("Create pizza with layers then add more layers") {
		val pizz = (create a pizza 
					spread sauce
					spread cheese
					toppings (onions, meat)
					spread cheese)
		assert("Pizza with 5 layers: sauce, cheese, onions, meat, cheese" == pizz.toString)
	}
}